system_metrics_collector/src/system_metrics_collector/main.cpp

```diff
@@ -85,6 +94,16 @@ int main(int argc, char ** argv)
         rcutils_get_error_string().str);
     }
   }
+  {
```

Comment:
I didn't review the previous one, but my rule of thumb is if you copied it 3 times, it's time to factor out into a function, e.g.
```
set_node_to_debug(mem_node, "memory");
set_node_to_debug(process_mem_node, "process memory");
...
```

However, why are we setting everything to DEBUG explicitly here? Shouldn't the externally configured ROS2 logger level be respected?

Reply:
Right now this main serves as a local manual tester for a linux system, in order to run each of the nodes and verify functionality. This will most likely go away once proper mains + launch files exist for the system cpu + memory and the process nodes are compose-able.

Reply:
"most likely go away" sounds a lot to me like "accidentally ends up being used forever" lol - should be trivial to make a function or macro to remove the copypasta

Reply:
Implemented the `set_node_to_debug()` function.